src/core/mesh/qgsmeshrenderersettings.cpp

```diff
@@ -340,6 +340,23 @@ void QgsMeshRendererVectorArrowSettings::readXml( const QDomElement &elem )
 
 // ---------------------------------------------------------------------
 
+QgsMeshRendererSettings::QgsMeshRendererSettings()
+  : mAveragingMethod( new QgsMeshSingleLevelAveragingMethod() )
+{
+}
+
+QgsMeshRendererSettings::~QgsMeshRendererSettings() = default;
+
+QgsMesh3dAveragingMethod *QgsMeshRendererSettings::averagingMethod() const
+{
+  return mAveragingMethod.get();
+}
+
+void QgsMeshRendererSettings::setAveragingMethod( QgsMesh3dAveragingMethod *method )
+{
+  mAveragingMethod.reset( method );
```

Comment:
would be good to add a safeguard ` if (method == mAveraginMethod.get()) return;`

Reply:
i will clone it here instead